server/app/routes/v1/stacks_api.rb

```diff
@@ -72,9 +100,17 @@ def deploy_stack(stack)
         end
 
         r.post do
-          r.on ':deploy' do
```

Comment:
Ouch, looks like a bug. The API would accept `POST /v1/stacks/:grid/:stack/anything` and handle it as a deploy?